### PR TITLE
Macos support

### PR DESCRIPTION
**Pull request by benoithudson**
I got the plugin working on macos with minimal changes; usdview can now display all my FBX cubes.

### CODE REVIEW

File: src/FbxNodeReader.cpp [hunk 1]
```diff
@@ -1400,8 +1400,14 @@ namespace
 				VtValue( std::move( prop.values ) ),
 				{ helpers::getDisplayGroupMetadata( UsdFbxDisplayGroupTokens->user ),
 				  { SdfFieldKeys->Custom, VtValue( true ) } } );
-			usdProp.timeSamples
-				= std::vector< std::tuple< UsdTimeCode, VtValue > >( prop.timeSamples.begin(), prop.timeSamples.end() );
+			for( const auto& timeSample : prop.timeSamples )
```
Comment (kminnaert) [inline src/FbxNodeReader.cpp]: What was the error exactly that CLANG gave you that pushed you towards this? The deleted lines should work, but that particular constructor is being deprecated after C++20 in favor of its `constexpr` sibling.

I would change this entirely loop to `usdProp.timeSamples.assign( prop.timeSamples.begin(), prop.timeSamples.end() )` instead. Or if all else fails, your basic copy assignment of `usdProp.timeSamples = prop.timeSamples` should also work.

On a side note, this loop assumes that `prop.timeSamples` is also a vector of `std::pair`, which it is not. It's a `std::tuple` so I'm actually surprised this compiled for you :)

**Review by kminnaert — CHANGES_REQUESTED** [review]
Looks straight forward enough on the CMAKE end, thanks for doing this!

File: src/FbxNodeReader.cpp (hunk 1 above)
Comment (benoithudson) [reply src/FbxNodeReader.cpp]: Tuple of 2 items and pair of the same two items convert implicitly, but you're right that's sloppy, I'll fix that.

I'm not sure how the original code compiled; one side's items are time and value, the other side has time and *list* of values. The exact error out of clang is about 10 pages long of course.

**Comment by benoithudson** [comment]
I removed the C++ changes.

I added cmake changes for Houdini.

Note that it doesn't immediately work on arm64 because Houdini requests macos version support that prevents usage of C++17 (I manually hack the generated .make file and remove the `-mmacosx-version-min=10.14` flag).

File: src/FbxNodeReader.cpp (hunk 1 above)
Comment (benoithudson) [reply src/FbxNodeReader.cpp]: I moved this to #3

**Review by kminnaert — APPROVED** [review]

**Pull request merged**